packages/common/src/api/tan-query/utils/primeUserData.ts

```diff
@@ -47,6 +48,11 @@ export const primeUserDataInternal = ({
       !queryClient.getQueryData(getUserQueryKey(user.user_id))
     ) {
       queryClient.setQueryData(getUserQueryKey(user.user_id), user)
+      if (users.length > 1) {
+        const ids = users.map((user) => user.user_id)
+        // TODO: this creates a second source of truth for the users
+        queryClient.setQueryData(getUsersQueryKey(ids), users)
```

Comment:
@dylanjeffers Ok so this alone doesn't quite solve the problem.

Given users `A, B, C` we are setting `['user', A], ['user', B], ['user', C]` and also `['users', 'ABC']` where each now has the full user object (or list of users) and is a source of truth for the data.
What we want is for `useUsers` to aggregate from the individual `'user'` slice where possible. So I think we're going to want the select based approach after all.